Deprecate 0.25 documentation (#1727)
astronomer/docs: software_versioned_docs/version-0.30/documentation-archive.md

Documentation sets for other maintained versions of Astronomer are updated to ensure accuracy and reflect existing functionality. This includes critical content fixes and updates based on bug fixes and minor features. These documentation sets, however, are not updated with enhanced content and quality-of-life improvements.

> Documentation sets for unmaintained versions of Astronomer Software no longer receive updates are not rendered on the Astronomer documentation site. You can still read these docs from the [Astronomer docs GitHub repo](https://github.com/astronomer/docs/blob/main/archived_software_docs).

Reviewer: ```suggestion
Documentation sets for unmaintained versions of Astronomer Software no longer receive updates and are not available on the Astronomer documentation site. Unmaintained documentation is available in the [Astronomer docs GitHub repository](https://github.com/astronomer/docs/blob/main/archived_software_docs).
```
